projects/Scraping Medium Articles/scraping_medium.py

```diff
@@ -0,0 +1,66 @@
+import os
+import sys
+import requests
+import re
+from bs4 import BeautifulSoup
+
+# switching to current running python files directory
+os.chdir('\\'.join(__file__.split('/')[:-1]))
+
+# function to get the html of the page
+def get_page():
+	url = input('Enter url of a medium article: ')
+	# handling possible error
+	if (url[:18] != 'https://medium.com') and (url[:17] != 'http://medium.com'):
+		print('Please enter a valid website, or make sure it is a medium article')
+		sys.exit(1)
+	res = requests.get(url)
+	res.raise_for_status()
+	soup = BeautifulSoup(res.text, 'html.parser')
+	return url, soup
+
+# function to remove all the html tags and replace some with specific strings
+def purify(text):
+    rep = {"<br>": "\n", "<br/>": "\n", "<li>":  "\n"}
+    rep = dict((re.escape(k), v) for k, v in rep.items()) 
+    pattern = re.compile("|".join(rep.keys()))
+    text = pattern.sub(lambda m: rep[re.escape(m.group(0))], text)
+    text = re.sub('\<(.*?)\>', '', text)
+    return text
+
+# function to compile all of the scraped text in one string
+def collect_text(url, soup):
+	fin = f'url: {url}\n\n'
+	main = (soup.head.title.text).split('|')
+	fin += f'Title:   {main[0].strip().upper()}\n{main[1].strip()}'
+
+	header = soup.find_all('h1')
+	j = 1
+
+	try:
+		fin += '\n\nINTRODUCTION\n'
+		for elem in list(header[j].previous_siblings)[::-1]:
+			fin += f'\n{purify(str(elem))}'
+	except:
+		pass
+
+	fin += f'\n\n{header[j].text.upper()}'
+	for elem in header[j].next_siblings:
+		if elem.name == 'h1':
+			j+=1
+			fin += f'\n\n{header[j].text.upper()}'
+			continue
+		fin += f'\n{purify(str(elem))}'
+	return fin
+
+# function to save file in the current directory
+def save_file(fin):
+	with open('scraped_article.txt', 'w', encoding='utf8') as outfile:
```

Comment:
instead of writing same file 'scraped_article.txt' use name from input url

Reply:
Okay sure i'll add that change

Reply:
I renamed the files according to the title of the article, and saved all of it to a folder 'scraped_articles'. Added some other changes accordingly. Can you please look into it?
And should i click on the 'resolve conversation' button now?